### PR TITLE
Tidy up typings a bit

### PR DESCRIPTION
**Pull request by azizhk**
Only 412 lines added. 1800 lines in yarn.lock due to integrity check.

### CODE REVIEW

File: src/components/Calendar.tsx [hunk 1]
```diff
@@ -23,17 +29,20 @@ class Calendar extends React.PureComponent<CalendarProps, CalendarState> {
...
-  private onChange = value => {
+  private onChange = (value: CalendarValue) => {
     const { range, onChange } = this.props;
     this.setState(
       {
         value
       },
-      () => (range ? value.length === 2 && onChange(value) : onChange(value))
+      () =>
+        range && Array.isArray(value)
```
Comment (azizhk) [inline src/components/Calendar.tsx]: this is logic change, make sure if range is true and value is not an array it should not call onChange(value)

File: src/components/Select.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ const Select: React.SFC<SelectProps> = props => {
     fullWidthDropdown
   } = props;
 
-  const OptionGroup: any = multiSelect ? OptionGroupCheckBox : OptionGroupRadio;
+  const OptionGroup = multiSelect ? OptionGroupCheckBox : OptionGroupRadio;
```
Comment (azizhk) [inline src/components/Select.tsx]: do in next PR

File: src/components/Calendar.tsx (hunk 1 above)
Comment (ritz078) [reply src/components/Calendar.tsx]: Can we have a test for this ? just this part.

**Comment by ritz078** [comment]
Tests and build are failing.

File: src/components/PhoneNumberInput.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { colors } from "../theme";
 export default class PhoneNumberInput extends React.Component<
   PhoneNumberInputProps
 > {
-  onCountrySelect = countryCode => {
+  onCountrySelect = (countryCode: string) => {
```
Comment (ritz078) [inline src/components/PhoneNumberInput.tsx]: React.ReactText

File: src/components/__tests__/dateinput.test.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+const date = new Date(2012, 11, 1);
+
+describe("DateInput", () => {
+  test("should return correct value on apply", () => {
+    const changeSpy = sinon.spy();
+
+    const dateinput = mount(
+      <DateInput placeholder="Select Date" onChange={changeSpy} value={date} />
+    );
+    dateinput.find(Input).simulate("click");
```
Comment (ritz078) [inline src/components/__tests__/dateinput.test.tsx]: camelcase

File: stories/slider.story.tsx [hunk 5]
```diff
@@ -9,16 +9,20 @@ const wrapperStyle = css({
...
+  value: [number, number];
+}
+
 storiesOf("Slider", module).add(
   "simple",
-  withState({ value: [10, 20] })(({ store }) => (
+  withState<State>({ value: [10, 20] })(({ store }) => (
     <Slider
-      onValuesUpdated={({ values }) => store.set({ value: values })}
+      onValuesUpdated={({ min, max }) => store.set({ value: [min, max] })}
```
Comment (azizhk) [inline stories/slider.story.tsx]: This was wrong.
Fixed in https://github.com/ritz078/pebble/pull/167/files#diff-221f70a125dc1788290163c421ba27a5R20

**Pull request merged**